### PR TITLE
Fixes long popup menu scroll behavior

### PR DESCRIPTION
**Pull request by timothyqiu**
Before this fix, popup menus longer than the viewport can be scrolled further than the top of the screen.

To reproduce:

1. Open the language menu in project manager
2. Scroll down (i.e. move the menu down)
    * The menu will move down until it gets out of the mouse focus.
    * The menu can be scrolled out of the viewport if you wish.
3. Scroll up, so the top of the menu is above the viewport
4. Scroll down again
    * The menu will not scroll further once the top of it is inside the viewport (as expected)
        * But the menu actually have one pixel outside the viewport (It has a two pixel border, but only one pixel is inside the viewport). If you scroll all the way up, the bottom of the menu is kept one pixel off the viewport too.

demo

### CODE REVIEW

**Comment by YeldhamDev** [comment]
I actually can't reproduce the bug to being with on master.

**Comment by timothyqiu** [comment]
Hi, I can still reproduce it on the current master (cd9d513).

Is the top border of the popup menu two pixel high, or one pixel?

Try making the window short first, and scroll down after opening the menu (If you scroll up first, then down, it won't happen).

**Comment by YeldhamDev** [comment]
I'm sorry, but still can't reproduce it.
I can, however, reproduce the "one pixel off" part.

**Comment by timothyqiu** [comment]
Maybe the popup menu's initial Y position is different between our machines.

On my machine, the initial Y position is 0 (so the menu initially has a two pixel top border.). The original code limits scrolling down only when `y < 0`, so I can scroll it down. 

It seems that the menu has an initial Y position of -1 then.

**Comment by YeldhamDev** [comment]
Just to be clear, my menu starts with the top still fully visible. Only after messing around with it that the top pixels stay off.

**Comment by timothyqiu** [comment]
That's strange.

Could you try that again with a touch pad? I don't have a mouse around.

**Comment by YeldhamDev** [comment]
Same behaviour with the touchpad.
Here's a GIF with how it behaves in my end (ignore the artifacts):
Peek 2019-12-11 11-12

**Comment by timothyqiu** [comment]
Hi, I just tested on a Windows machine and I can't reproduce the main issue.

For macOS, I tried scrolling with a mouse, can't reproduce either. But I can reproduce that with two finger scrolls using the touch pad.

The scroll-down part might just be on macOS then. I'm sorry for didn't test those before.

The one pixel off part applies to all platforms I think.

**Comment by timothyqiu** [comment]
After further investigation, I can confirm that only macOS can scroll the menu further down. macOS seems to be the only OS that emits the `InputEventPanGesture` event in Godot (by scrolling with the trackpad), and `InputEventPanGesture` scroll and mouse wheel scroll are handled differently.

Anyway, let me rephrase the issue:

1. Once the top / bottom border of a long popup menu is outside the viewport, you can't scroll it back inside. There is always at least a one pixel offset.
2. On macOS, if you scroll a long popup menu down using the trackpad pan gesture after popup, it can be scrolled down even after the top of the menu is already inside the viewport.

I thought these two are related issues, but I'm wrong. However, fixing the first issue did resolve the second one. Because to get rid of the 1 pixel offset, the scroll logic has to handle the `y == 0` case.

**Comment by akien-mga** [comment]
Thanks!
